juju/juju
# state: Add UserConnections to the all watcher

Add all watcher events for notifying on user connections. This implements
the feature requested in https://bugs.launchpad.net/juju/+bug/1929524.

## Checklist


## QA steps

 1. Create a controller with this code version.
 2. Connect a client that calls WatchAllModels on the Controller facade and displays the results.
 3. Connect to a model on the controller (juju status)
 4. Verify the event stream includes a userConnection event with the connection time around step 3.

## Documentation changes

This adds a new event type into the AllWatcher event stream. The "userConnection" events will be fired every time a user connects to a model.

## Bug reference

 - https://bugs.launchpad.net/juju/+bug/1929524
